unit_tests/testParameterLists/testParameterLists.f90

```diff
@@ -3765,6 +3773,886 @@ SUBROUTINE testSDKa3()
 ENDSUBROUTINE testSDKa3
 !
 !-------------------------------------------------------------------------------
+!Test 4-D Array SNK support
+SUBROUTINE testSNKa4()
+  LOGICAL(SBK) :: bool
+  CHARACTER(LEN=EXCEPTION_MAX_MESG_LENGTH) :: msg,refmsg
+  ALLOCATE(valsnka4(2,2,2,2))
+  valsnka4(1,1,1,:)=1_SNK
+  valsnka4(2,1,1,:)=2_SNK
+  valsnka4(1,2,1,:)=3_SNK
+  valsnka4(2,2,1,:)=4_SNK
+  valsnka4(1,1,2,:)=5_SNK
+  valsnka4(2,1,2,:)=6_SNK
+  valsnka4(1,2,2,:)=7_SNK
+  valsnka4(2,2,2,:)=8_SNK
+  !test init
+
+  COMPONENT_TEST('%init(...)')
+  ASSERT(.NOT.ASSOCIATED(testParam%pdat),'%pdat 1')
+  ASSERT(LEN(testParam%name) == 0,'%name 1')
```

Comment:
Use `ASSERT_EQ`?

Reply:
Yeah... once I figure out what's up with the build error I plan to just search/replace as many of these ASSERT to ASSERT_EQ/ASSERT_APPROXEQ as I can

Reply:
I've decided not to do this right now... there are a LOT of things that can't be passed through the ASSERT_EQ interfaces because they're using an == operator that we wrote that isn't recognized during the preprocessing step.  I'm not clear on why it matters, but it apparently does. And with the unit test being nearly 7000 lines long I don't have time to actually look at every single replacement I'm doing.